### PR TITLE
Default argument names

### LINKED ISSUE #799: Use default_ argument names
## Description

Use `default_<arg_name>` in `__init__` where `<arg_name>` matches the name of a type parameter of the operation.

## Motivation and context

In classes such as the MD pair potentials. HPMC integrators, and possibly other places use the same name in the default value argument as the type parameter. This fluid representation is confusing to users and is not explicit. See the corresponding discussion in #746

### PR DESCRIPTION
**Pull request by seoulfood**
## Description

Renamed arguments in __init__ subroutines to have default_<argname> so as to make this distinct from the type parameter.

In md `r_cut` and `r_on` are changed to `default_r_cut` and `default_r_on`. In hpmc the maximal step size `d` and the maximal rotation `a` to `default_d` and `default_a` respectively. 

## Motivation and context

To provide clarity to users and provide distinction between initialization assignment and type assignment for various parameters.

Resolves #799

## How has this been tested?

Updated pytest files and ran these tests.

We checked the sphinx docs, and they also display properly.

## Change log

```
Properly named default values to resolve confusion between initialization and type assignment.
```

## Checklist:

- [X] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [X] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [X] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/md/many_body.py [hunk 1]
```diff
@@ -33,22 +33,22 @@ class Triplet(Force):
 
         \begin{eqnarray}
         \vec{F_i}
-            =& -\nabla V(\vec r_{ij}, \vec r_{ik})
-             & r_{ij} < r_{\mathrm{cut}}
-                \textrm{ and } r_{ik} < r_{\mathrm{cut}} \\
+            =& -\nabla V(\vec default_r_{ij}, \vec default_r_{ik})
+             & default_r_{ij} < default_r_{\mathrm{cut}}
+                \textrm{ and } default_r_{ik} < default_r_{\mathrm{cut}} \\
```
Comment (b-butler) [inline hoomd/md/many_body.py:38]: This should not be changed.

File: hoomd/md/many_body.py [hunk 2]
```diff
@@ -33,22 +33,22 @@ class Triplet(Force):
...
            =& 0 & else
         \end{eqnarray}
 
     Where
-    :math:`\vec r_{\alpha \beta} = \vec r_{\alpha} - \vec r_{\beta}`.
+    :math:`\vec default_r_{\alpha \beta} = \vec default_r_{\alpha} - \vec default_r_{\beta}`.
```
Comment (b-butler) [inline hoomd/md/many_body.py:43]: Should not be changed.

File: hoomd/md/many_body.py [hunk 3]
```diff
@@ -149,10 +149,10 @@ class Tersoff(Triplet):
         \begin{equation}
             f_C(r) =
             \begin{cases}
-                1 & r < r_{\mathrm{cut}} - r_{CT} \\
+                1 & r < default_r_{\mathrm{cut}} - default_r_{CT} \\
                 \exp \left[-\alpha\frac{x(r)^3}{x(r)^3 - 1} \right]
-                  & r_{\mathrm{cut}} - r_{CT} < r < r_{\mathrm{cut}} \\
-                0 & r > r_{\mathrm{cut}}
+                  & default_r_{\mathrm{cut}} - default_r_{CT} < r < default_r_{\mathrm{cut}} \\
+                0 & r > default_r_{\mathrm{cut}}
```
Comment (b-butler) [inline hoomd/md/many_body.py:155]: Should not change

File: hoomd/md/many_body.py [hunk 4]
```diff
@@ -165,7 +165,7 @@ class Tersoff(Triplet):
 
     .. math::
 
-        x(r) = \frac{r - (r_{\mathrm{cut}} - r_{CT})}{r_{CT}}
+        x(r) = \frac{r - (r_{\mathrm{cut}} - default_r_{CT})}{r_{CT}}
```
Comment (b-butler) [inline hoomd/md/many_body.py:168]: Should not change.

File: hoomd/md/many_body.py [hunk 5]
```diff
@@ -296,9 +296,9 @@ class RevCross(Triplet):
             \begin{eqnarray*}
             \hat{v}^{ \left( 2b \right)}_{ij}\left(\vec{r}_{ij}\right) =
             \begin{cases}
-            1 & r \le r_{min} \\
+            1 & r \le default_r_{min} \\
             - \dfrac{v_{ij}\left(\vec{r}_{ij}\right)}{\epsilon}
-              & r > r_{min} \\
+              & r > default_r_{min} \\
             \end{cases}
```
Comment (b-butler) [inline hoomd/md/many_body.py:302]: No change

File: hoomd/md/many_body.py [hunk 6]
```diff
@@ -406,15 +406,15 @@ class SquareDensity(Triplet):
...
+        w(x) = \frac{15}{2 \pi default_r_{c,\mathrm{weight}}^3}
                (1-r/r_{c,\mathrm{weight}})^2
 
     The local density at the location of particle *i* is defined as
 
     .. math::
 
         n_i = \sum\limits_{j\neq i} w_{ij}
-              \left(\big| \vec r_i - \vec r_j \big|\right)
+              \left(\big| \vec default_r_i - \vec default_r_j \big|\right)
```
Comment (b-butler) [inline hoomd/md/many_body.py:417]: No change

**Review by b-butler — CHANGES_REQUESTED** [review]
Thanks for the PR! :smiley:. `hoomd/md/special_pair.py` needs to be refactored with `r_cut` as well and some math needs to be reverted.

File: hoomd/md/pair/aniso.py [hunk 7]
```diff
@@ -65,8 +65,7 @@ class Dipole(AnisotropicPair):
 
     Args:
         nlist (`hoomd.md.nlist.NList`): Neighbor list
-        r_cut (float): Default cutoff radius (units: [length]).
-        r_on (float): Default turn-on radius (units: [length]).
+        default_r_cut (float): Default cutoff radius (units: [length]).
```
Comment (Charlottez112) [inline hoomd/md/pair/aniso.py:68]: Why are we not changing `r_on` ? And related to this question: @b-butler could you remind me why only `r_cut` is the valid argument shown in the documentation but not `r_on`? (in the doc: `class hoomd.md.pair.aniso.Dipole(nlist, r_cut=None, mode='none')`)

File: hoomd/md/pair/aniso.py [hunk 8]
```diff
@@ -147,8 +146,7 @@ class GayBerne(AnisotropicPair):
 
     Args:
         nlist (`hoomd.md.nlist.NList`): Neighbor list
-        r_cut (float): Default cutoff radius (units: [length]).
-        r_on (float): Default turn-on radius (units: [length]).
```
Comment (Charlottez112) [inline hoomd/md/pair/aniso.py:151]: See comment above (in the doc `class hoomd.md.pair.aniso.GayBerne(nlist, r_cut=None, mode='none')`)

File: hoomd/md/many_body.py [hunk 9]
```diff
@@ -174,7 +174,7 @@ class Tersoff(Triplet):
     .. math::
 
         \chi_{ij} = \sum_{k \neq i,j} f_C(r_{ik})
-                     \cdot e^{\lambda_3^3 |r_{ij} - r_{ik}|^3}
+                     \cdot e^{\lambda_3^3 |r_{ij} - default_r_{ik}|^3}
         \cdot g(\theta_{ijk})
```
Comment (Charlottez112) [inline hoomd/md/many_body.py:178]: `r` in math (see @b-butler's comment)

File: hoomd/md/pair/aniso.py (hunk 7 above)
Comment (b-butler) [reply hoomd/md/pair/aniso.py:68]: `r_on` is not in the constructor, but squeaked through the documentation somehow. This is just removing it. `r_on` does not exist for the anisotropic potentials.

File: hoomd/md/pair/aniso.py (hunk 8 above)
Comment (b-butler) [reply hoomd/md/pair/aniso.py:151]: The removal of `r_on` is correct.

**Comment by joaander** [comment]
I'm resolving merge conflicts on this branch now.

**Comment by joaander** [comment]
Sorry, I hijacked this PR with a few unrelated changes that become obvious when I solved the merge conflicts.

**Comment by joaander** [comment]
The F401 flake8 errors are coming from `master` and will be solved when #1030 merges.

**Review by b-butler — COMMENTED** [review]
In general, LGTM. However, before I approve should we just remove `mode` for those pair potentials that have no non-none modes? `ParameterDict` objects support `__delitem__`.

**Comment by joaander** [comment]
> In general, LGTM. However, before I approve should we just remove `mode` for those pair potentials that have no non-none modes? `ParameterDict` objects support `__delitem__`.

We could. Wouldn't users expect it to be there based on the `Pair` attributes documentation, though?

**Comment by joaander** [comment]
Odd, now the tests are seg faulting.

**Comment by joaander** [comment]
I reproduced it locally. In `PotentialPairDPDThermo`, `m_T` is a null pointer:
```
* thread #2, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x0000000120d8d524 _md.cpython-39-darwin.so`PotentialPairDPDThermo<EvaluatorPairDPDThermo>::computeForces(this=0x000000010099ea00, timestep=0) at PotentialPairDPDThermo.h:216:40
   213              evaluator eval(rsq, rcutsq, param);
   214 
   215              // Special Potential Pair DPD Requirements
-> 216              const Scalar currentTemp = (*m_T)(timestep);
   217 
   218              // set seed using global tags
   219              unsigned int tagi = h_tag.data[i];
Target 0: (Python) stopped.
(lldb) print m_T
(std::shared_ptr<Variant>) $0 = nullptr {
  __ptr_ = nullptr
}
```
I don't know why.

**Pull request merged**